externals/kyuubi-flink-sql-engine/src/main/java/org/apache/kyuubi/engine/flink/context/ExecutionContext.java

```diff
@@ -0,0 +1,330 @@
+
+    // Must initialize the table engineEnvironment before actually the
+    createTableEnvironment(settings, config, catalogManager, moduleManager, functionCatalog);
+
+    // No need to register the catalogs if already inherit from the same session.
+    initializeCatalogs();
+  }
+
+  private void createTableEnvironment(
+      EnvironmentSettings settings,
+      TableConfig config,
+      CatalogManager catalogManager,
+      ModuleManager moduleManager,
+      FunctionCatalog functionCatalog) {
+    if (engineEnvironment.getExecution().isStreamingPlanner()) {
+      streamExecEnv = createStreamExecutionEnvironment();
+      execEnv = null;
+
+      final Map<String, String> executorProperties = settings.toExecutorProperties();
+      executor = lookupExecutor(executorProperties, streamExecEnv);
+      tableEnv =
+          createStreamTableEnvironment(
+              streamExecEnv,
+              settings,
+              config,
+              executor,
+              catalogManager,
+              moduleManager,
+              functionCatalog);
+    } else if (engineEnvironment.getExecution().isBatchPlanner()) {
+      streamExecEnv = null;
+      execEnv = createExecutionEnvironment();
+      executor = null;
+      tableEnv = new BatchTableEnvironmentImpl(execEnv, config, catalogManager, moduleManager);
+    } else {
+      throw new RuntimeException("Unsupported execution type specified.");
+    }
+  }
+
+  private void initializeCatalogs() {
+    // --------------------------------------------------------------------------------------------------------------
+    // Step.6 Set current catalog and database.
+    // --------------------------------------------------------------------------------------------------------------
+    // Switch to the current catalog.
+    Optional<String> catalog = engineEnvironment.getExecution().getCurrentCatalog();
+    catalog.ifPresent(tableEnv::useCatalog);
+
+    // Switch to the current database.
+    Optional<String> database = engineEnvironment.getExecution().getCurrentDatabase();
+    database.ifPresent(tableEnv::useDatabase);
+  }
+
+  private ExecutionEnvironment createExecutionEnvironment() {
+    final ExecutionEnvironment execEnv = ExecutionEnvironment.getExecutionEnvironment();
+    execEnv.setRestartStrategy(engineEnvironment.getExecution().getRestartStrategy());
+    execEnv.setParallelism(engineEnvironment.getExecution().getParallelism());
+    return execEnv;
+  }
+
+  private StreamExecutionEnvironment createStreamExecutionEnvironment() {
+    final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+    env.setRestartStrategy(engineEnvironment.getExecution().getRestartStrategy());
+    env.setParallelism(engineEnvironment.getExecution().getParallelism());
+    env.setMaxParallelism(engineEnvironment.getExecution().getMaxParallelism());
+    env.setStreamTimeCharacteristic(engineEnvironment.getExecution().getTimeCharacteristic());
```

Comment:
deprecated method

Reply:
EventTime is the default config option. so remove it.